### PR TITLE
Improve clangrt lib path normalization.

### PR DESCRIPTION
**Pull request by mjburghard**
Improvements as suggested by @keith [here](https://github.com/bazelbuild/rules_apple/pull/1905#discussion_r1140477111).

### CODE REVIEW

File: tools/clangrttool/clangrttool_unittest.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import os
+import unittest
+
+from tools.clangrttool.clangrttool import normalize_clang_lip_path, ClangRuntimeToolError
+
+clang_lib_path_common = "Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/lib/darwin"
+xcode_developer_dir_default = "/Applications/Xcode.app/Contents/Developer"
```
Comment (brentleyjones) [inline tools/clangrttool/clangrttool_unittest.py:7]: Can we add one that is named something different, like `ProjectBuilder.app`?

File: tools/clangrttool/clangrttool.py [hunk 2]
```diff
@@ -25,6 +25,21 @@
...
 
+def normalize_clang_lip_path(path, local_developer_dir):
+  normalized_path_components = os.path.normpath(path).split(os.path.sep)
+
+  path_index = None
+  for i in range(len(normalized_path_components) - 1):
+      if (normalized_path_components[i] == "Contents" and
+        normalized_path_components[i+1] == "Developer"):
+        path_index = i
+        break
```
[the remark this answers is not in the document]
Comment (mjburghard) [reply tools/clangrttool/clangrttool.py:36]: Just checked, the using a regex replacement looks simpler and should be easier to maintain. Will switch to it.

File: tools/clangrttool/clangrttool_unittest.py (hunk 1 above)
Comment (mjburghard) [reply tools/clangrttool/clangrttool_unittest.py:7]: See line 12 below.

**Review by brentleyjones — APPROVED** [review]

File: tools/clangrttool/clangrttool.py [hunk 3]
```diff
@@ -21,10 +21,17 @@
 """
 
 import os
+import re
 import subprocess
 import sys
 import zipfile
 
+def normalize_clang_lip_path(path, local_developer_dir):
+  return re.sub(".*/Contents/Developer",
```
Comment (keith) [inline tools/clangrttool/clangrttool.py:30]:
```suggestion
  return re.sub(".*\.app/Contents/Developer",
```

marginally safer, no need to take it if you don't want to

**Review by keith — APPROVED** [review]
is this ready to go?

File: tools/clangrttool/clangrttool.py (hunk 3 above)
Comment (mjburghard) [reply tools/clangrttool/clangrttool.py]: I'll take it.
And yes, ready to go.

**Comment by keith** [comment]
so far the CI failure is an infra issue, it's automatically retrying slowly

**Pull request merged**